### DISCUSSION TITLE
Multiple builds from one app dir

### DISCUSSION THREAD

**Discussion by andershoegh** [vercel/next.js#59692] [Help]
### Summary

Any help appreciated on this topic!

I have the requirement to build two applications in nextjs, with the app directory. The first application is called "platform" the other "tool". Platform will deploy to the web.  Tool will be included in a CLI and used locally. Eventually Tool will be integrated and used inside the Platform application, where users on the platform will have access to the functionality of Tool. Tool will always need to be deployed without Platform to a local environment.

What would be the recommended way of approaching this requirement? Is it possible to have multiple builds from one app dir, or would it be better to have two NextJS repositories and implement Tool into Platform through a microfrontend solution or something similar. Preferably our solution would enable us to share UI components across the two applications too.

Are there other ways to handle this use case that would be better?

**Comment by janekolszak** [comment]
I had the same problem: https://stackoverflow.com/questions/77890970/multiple-next-config-js-configuration-files-in-next-js
